src/GridFS/Bucket.php

```diff
@@ -551,6 +551,7 @@ private function getRawFileDocumentForStream($stream)
         if ( ! isset ($metadata['wrapper_data']) || ! $metadata['wrapper_data'] instanceof StreamWrapper) {
             throw InvalidArgumentException::invalidType('$stream wrapper data', isset($metadata['wrapper_data']) ? $metadata['wrapper_data'] : null, 'MongoDB\Driver\GridFS\StreamWrapper');
         }
+        $metadata['wrapper_data']->stream_close();
```

Comment:
You're right, it's probably not a good idea.
I first tried to modify the implementation of `WritableStream::getFile()` to return the `size` property.
```php
    public function getFile()
    {
        $this->file['size'] = $this->getSize();

        return (object) $this->file;
    }
```
but in my case the size is not up to date : 
```php
$stream = $bucket->openUploadStream($filename, $options);
fwrite($stream, $string);
$fileDocument = $bucket->getFileDocumentForStream($stream);
```
@jmikola Do you see another way to recover the up-to-date size ?